### ISSUE TITLE
Code reformat breaks the code

### ISSUE THREAD

**Issue by drhumlen** [JetBrains/fsharp-support#58]
When running code `Reformat Code` (cmd+opt+L), the produced code is broken because of indentation when using [<Annotations>]:

```F#
[<CLIMutable>]
type CreateUser =
    { [<JsonRequired>]
      Name: string }
```

is changed to:

```F#
[<CLIMutable>]
type CreateUser =
    {  [<JsonRequired>]
      Name: string }
```

^ Notice the extra space in front of [<JsonRequired>]. That make the code on the next line "off side", which causes the whole parsing to fail.

----
Also, one unrelated question: I read on fantomas' site that Rider uses it for its formatting. However, fantomas' formatting is _way_ more opinionated than Rider's. Imo that's mostly a good thing, but it would be nice if it was configurable so that I could choose fantomas' more opinionated formatting if I wanted to.

**Comment by nojaf** [comment]
Hey, Rider uses Fantomas so if there is a problem you should report in https://github.com/fsprojects/fantomas/.

Could you try this out in the online tool? https://jindraivanek.gitlab.io/fantomas-ui/#?fantomas=preview-trivia
We are working on a new version of Fantomas and some things might already be fixed.
If you encounter an issue in the tool (using the latest version, vNext 3.X), please create an issue using the create issue link below.

**Comment by drhumlen** [comment]
@nojaf : Trying the code above in the online tool you linked gives _correct_ code. When I do the same in Rider, it gives the _wrong_ code. So it seems to have been fixed in Fantomas then I guess :D  so unless the error is in Rider, it will probably start working again when you roll out your new version.

Is there an ETA on that? Also, can you add formatting options to the (fantomas) code formatting in Rider? Fantomas has a ton of options, and Rider seems to deviate quite far from Fantomas' default settings.

**Comment by auduchinok** [comment]
> @nojaf : Trying the code above in the online tool you linked gives correct code. When I do the same in Rider, it gives the wrong code.

This could be due to Rider uses different default settings compared to the online tool. We have `PreserveEndOfLine` enabled by default to try to preserve used formatting style where possible.

> Is there an ETA on that? Also, can you add formatting options to the (fantomas) code formatting in Rider? Fantomas has a ton of options

All Fantomas options are already in `File | Settings | Editor | Code Style | F#`.

**Comment by drhumlen** [comment]
@auduchinok : Ah. Nice. I tried to search for "fantomas" in the settings without finding anything, so that's why I assumed the settings wasn't there. But that's the fantomas settings. I see. Nice 😄 

> This could be due to Rider uses different default settings compared to the online tool. We have PreserveEndOfLine enabled by default to try to preserve used formatting style where possible.

I tried removing `PreserveEndOfLine` now, and now none of my code breaks when I hit `Reformat Code` 😄. I don't always agree with Fantomas (it formats certain code in a weird way), but it's nice to have the option to see how Fantomas' would've formatted my code by comparison.

**Comment by rulrok** [comment]
Hello, I'm experiencing this annoying issue as well.

I created a [bug on Rider bug track](https://youtrack.jetbrains.com/issue/RIDER-33500) but they asked to go on `fantomas` project.

There, I found that the [bug had been fixed already](https://github.com/fsprojects/fantomas/issues/442).

I'd like that the formatting would work without having to change the `PreserveEndOfLine` option.

I prefer having:

```
    and SubscriptionSuspendedEvent = {
        SubscriptionId: SubscriptionId
        AccountId: string
    }
```

rather than (`PreserveEndOfLine` disabled):

```
    and SubscriptionSuspendedEvent =
        { SubscriptionId: SubscriptionId
          AccountId: string }
```

**Comment by drhumlen** [comment]
Maybe @nojaf can answer? ^

@rulrok I too prefer the former style. While the second style is prettier in many ways, the former is more practical and while e.g. result in less git diff when add another field. It's almost more similar to how you would define types in other languages like TypeScript.

@nojaf Maybe there should be an option for formatting types & lists in the former style? Or has that already been discussed somewhere?

**Comment by nojaf** [comment]
Hey, this looks like https://github.com/fsprojects/fantomas/issues/453

**Comment by auduchinok** [comment]
Fixed in 2019.3 with Fantomas update.
